app/models/role.rb

```diff
@@ -32,4 +32,10 @@ class Role < ApplicationRecord
             inclusion: { in: ROLES }
 
   scopify
+
+  def resource_name
+    return resource_type unless resource_id
+
+    Tag.find(resource_id).name
```

Comment:
Can you explain this method? I don't get it 😭 
We return the resource type if there is no resource id but if there is a resource id we return the tag name? But `resource_type` can be:

```ruby
[1] pry(main)> Rolify.resource_types
=> ["LiquidTags::UserSubscriptionTag", "User"]
```

I tried to read this a couple of times out loud but it's still not clear to me, why do we load tags if it can be anything else?

Reply:
It's been a couple of days since I wrote this, so I have to think! :laughing: 

My thought was that many Roles have a `resource_type` (specifically, I was thinking of `single_resource_admin`s), and that type constitutes a resource "name": 

```ruby
# In the case of a single resource admin, where resource_id is always nil
# but resource_type indicates which model is associated with this admin
User.last.add_role(:single_resource_admin, Article)
User.last.roles.first.resource_name #=> "Article"

# In the case of a tag moderator (using one I made via the UI)
u.roles.last.resource_name #=> "python"
```

It seems like I may have missed a use case? Or does this make sense to you?

Reply:
Got it! My suggestion then is to add it this explainer as a comment if possible :)

Reply:
Comment added!